### PR TITLE
Account in cura compatibility

### PR DESCRIPTION
**Pull request by ChrisTerBeke**
Change from using internal auth system to Cura's new account API.

### CODE REVIEW

**Comment by ChrisTerBeke** [comment]
@nallath To finish this I'm missing:

* A property/method to get the access token from the account API
* The logged_in state as first argument in the loginStateChanged signal callback

File: CuraDrive/src/DrivePluginExtension.py [hunk 1]
```diff
@@ -118,11 +107,8 @@ def _storeBackupDate(self) -> None:
         backup_date = datetime.now().strftime(self.DATE_FORMAT)
         self._preferences.setValue(Settings.AUTO_BACKUP_LAST_DATE_PREFERENCE_KEY, backup_date)
 
-    def _onLoginStateChanged(self, logged_in: bool = False, error_message: str = None):
+    def _onLoginStateChanged(self, logged_in: bool = False):
```
Comment (nallath) [inline CuraDrive/src/DrivePluginExtension.py]: missing return type

File: CuraDrive/src/DriveApiService.py [hunk 2]
```diff
@@ -150,7 +144,7 @@ def deleteBackup(self, backup_id: str) -> bool:
         :param backup_id: The ID of the backup to delete.
         :return: Success bool.
         """
-        access_token = self._getAccessToken()
+        access_token = self._cura_api.account.accessToken
```
[the remark this answers is not in the document]
Comment (DanielSchiavini) [reply CuraDrive/src/DriveApiService.py]: Then I suggest we make a generic scope that represents being logged in in Cura, and not specific ones like `packages.package.download` as suggested in https://confluence.ultimaker.com:8443/display/SD/0009+-+OAuth2+scopes+improvements
Cura needs to request the scopes in the OAuth authorize endpoint, so it cannot be changed without new releases.

[the remark this answers is not in the document]
Comment (ChrisTerBeke) [reply CuraDrive/src/DriveApiService.py]: That sounds like a good idea. Then endpoints in the cloud that need to support Cura can just accept that scope.

**Comment by ChrisTerBeke** [comment]
I'll merge once the Cura changes are merged as well. Then we can make an experimental build of the plugin for internal use.

File: CuraDrive/src/DriveApiService.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChrisTerBeke) [reply CuraDrive/src/DriveApiService.py]: We discussed this and decided to stay at the current situation (feature-based scopes) as adding a new scope usually requires functional changes in Cura anyways.

**Pull request merged**